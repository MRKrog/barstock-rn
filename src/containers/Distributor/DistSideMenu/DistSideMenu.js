import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './DistSideMenu.style';
import { NavigationActions } from 'react-navigation';
import { ScrollView, Text, View, TouchableOpacity } from 'react-native';


class DistSideMenu extends Component {
  navigateToScreen = (route) => () => {
    const navigateAction = NavigationActions.navigate({
      routeName: route
    });
    let { routeName } = this.props.navigation.state;
    console.log(this.props.navigation.state.routes);

    this.props.navigation.dispatch(navigateAction);
  }

  render () {
    return (
      <View style={styles.container}>
        <View>
          <Text style={styles.sectionHeadingStyle}></Text>
          <View style={styles.navSectionStyle}>
            <TouchableOpacity style={styles.navItemContainer} onPress={this.navigateToScreen('Login')}>
              <Text style={styles.navItemStyle}>Logout</Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    );
  }
}

DistSideMenu.propTypes = {
  navigation: PropTypes.object
};

export default DistSideMenu;
