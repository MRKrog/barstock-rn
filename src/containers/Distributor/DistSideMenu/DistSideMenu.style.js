export default {
  container: {
    display: "flex",
    justifyContent: "space-between",
    paddingTop: 20,
    flex: 1,
    backgroundColor: '#231f20'
  },
  navItemContainer: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "flex-start",
    alignItems: "center",
    paddingHorizontal: 20,
    paddingVertical: 20,
  },
  iconStyle: {
    marginRight: 15
  },
  navItemStyle: {
    fontSize: 18,
    textTransform: "uppercase",
    fontFamily: 'abel',
    textAlign: "left",
    color: "#ffffff",
    letterSpacing: 1,
  },

};
