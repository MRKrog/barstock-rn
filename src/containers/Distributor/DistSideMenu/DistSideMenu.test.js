import { shallow } from "enzyme"
import DistSideMenu from "./DistSideMenu"
import React from "react"

describe("DistSideMenu", () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <DistSideMenu />
    )
  })

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  })
})