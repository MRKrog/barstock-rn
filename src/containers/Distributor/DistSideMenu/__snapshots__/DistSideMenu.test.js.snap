// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistSideMenu should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#231f20",
      "display": "flex",
      "flex": 1,
      "justifyContent": "space-between",
      "paddingTop": 20,
    }
  }
>
  <View>
    <Text />
    <View>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="#ffffff"
          name="sign-out"
          size={20}
          style={
            Object {
              "marginRight": 15,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "abel",
              "fontSize": 18,
              "letterSpacing": 1,
              "textAlign": "left",
              "textTransform": "uppercase",
            }
          }
        >
          Logout
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
